File: Python_Solutions/ValidParenthesis.py
class Solution:
    def isValid(self, s: str) -> bool:
        map = {')':'(', '}':'{', ']':'['}
        stack = []
        for character in s:
            if character in map.values():
                stack.append(character)
            else:
                if len(stack) >= 1 and stack[-1] == map[character]:
                    stack.pop()
                else:
                    return False
        return len(stack) == 0

File: Java_Solutions/ValidParenthesis.java
class Solution {
    public boolean isValid(String s) {
        HashMap<Character, Character> bracketsMap = new HashMap<>();
        bracketsMap.put(')','(');
        bracketsMap.put(']','[');
        bracketsMap.put('}','{');
        Stack<Character> bracketStack = new Stack<Character>();

        for (char bracket : s.toCharArray()){
            if (bracketsMap.containsValue(bracket)){
                bracketStack.push(bracket);
            }
            else{
                if (bracketStack.size() >=1 && bracketStack.peek() == bracketsMap.get(bracket)) {
                    bracketStack.pop();
                }
                else{
                    return false;
                }
            }
        }
        return bracketStack.size() == 0;
    }
}